doc/index.rst

```diff
@@ -124,3 +124,4 @@ Indices and tables
    python3
    migrate-to-pymongo4
    developer/index
+   common-issues
```

Comment:
This tree is hidden. We  need to add a link to the new page in the "Overview" section otherwise it doesn't show up: https://pymongo--918.org.readthedocs.build/en/918/#overview